Assets/Scripts/UI/TextBox.cs

```diff
@@ -0,0 +1,70 @@
+﻿using UnityEngine;
+using System.Collections;
+using UnityEngine.UI;
+
+public class TextBox : MonoBehaviour {
+
+	Text _UIText = null;
+
+	[SerializeField] KeyCode _clearKey = 0;
+
+	Coroutine _textFade;
+
+	void Start ()
+	{
+		_UIText = GetComponent<Text>();
+	}
+
+	void Update()
+	{
+		if ( Input.GetKeyDown( _clearKey ) )
+		{
+			ClearText();
+			CancelInvoke( "ClearText" );
+		}
+	}
+
+	public void SetText( string textString )
+	{
+		ClearText();
+
+		_UIText.text = textString;
+	}
+	
+	public void SetTextForDuration( string textString, float duration = 5.0f, float fadeoutDuration = 2.0f )
+	{
+		ClearText();
+
+		_UIText.text = textString;
+		_textFade = StartCoroutine( Fadeout( duration, fadeoutDuration ) );
+	}
+
+	public void ClearText()
+	{
+		_UIText.text = "";
+
+		if ( _textFade != null )
+		{
+			StopCoroutine( _textFade );
+		}
```

Comment:
I'm guessing it's automatic, though on that note I should remove the line in the coroutine that does null it out.